tool/tctl/common/auth_command.go

```diff
@@ -879,6 +904,22 @@ func (a *AuthCommand) checkProxyAddr(clusterAPI auth.ClientI) error {
 	return trace.BadParameter("couldn't find registered public proxies, specify --proxy when using --format=%q", identityfile.FormatKubernetes)
 }
 
+func parseURL(rawurl string) (*url.URL, error) {
+	u, err := url.Parse(rawurl)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	// If no scheme is provided url.Parse fails the parsing and considers the host
+	// as scheme, leaving the host empty.
+	if u.Host == "" {
+		return &url.URL{
+			Host: rawurl,
+		}, nil
+	}
```

Comment:
Is this only ever going to be called on `scheme://host:port`-like strings?

Reply:
This code should only be called if they used the proxy flag without scheme, i.e. `--proxy=tele.teleport.local:3080`.
Golang `url.Parse` has a weird behavior when parsing scheme-less URLs.
https://go.dev/play/p/Upa1vDCXTiM

I can explode the `rawURL` to ensure we do not include `/` and that the `host` is correctly defined if you think it's necessary.

Reply:
I had no idea we supported `http://host:port` and `https://host:port` there.

Maybe if parsing as a URL results in a blank schema we should `SplitHostPort` instead?

Reply:
Before this PR, you were forced to include the `http(s)://` because the `url.Parse` generated a wrong URL and returned an error [here](https://github.com/gravitational/teleport/pull/20497/files#diff-1da7bc0f8a6f715d9554a2bf8b7a97551d546cb187ab2135b78df3cf7636f21dL844).

It's not required that you define the `:port` portion, so  `SplitHostPort` will fail for those cases.
Cloud exposes port `:443` and some examples don't even mention the port.
[one of them](https://goteleport.com/docs/database-access/reference/cli/?scope=cloud)